### PR TITLE
docs(cli): add configuration section

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/CDX-919

## Proposed changes
Add some documentation about the config
main source: https://oclif.io/docs/config#docsNav

### CODE REVIEW

**Review by olamothe — APPROVED** [review]

File: packages/cli/core/README.md [hunk 1]
```diff
@@ -31,6 +31,28 @@ USAGE
 
 <!-- usagestop -->
 
+# Configuration
+
+The Coveo CLI is based on oclif and use the following paths:
```
Comment (jpmarceau) [inline packages/cli/core/README.md:36]:
```suggestion
The Coveo CLI is based on oclif and uses the following paths:
```

**Review by jpmarceau — APPROVED** [review]
👍

File: packages/cli/core/README.md [hunk 2]
```diff
@@ -31,6 +31,28 @@ USAGE
...
+
+- **cacheDir** - CLI cache directory
+  - macOS: `~/Library/Caches/mycli`
```
Comment (y-lakhdar) [inline packages/cli/core/README.md:39]: Why `mycli`? Shouldn't be `@coveo/cli`

Comment (louis-bompart) [reply packages/cli/core/README.md:39]: oh yeah. I went a bit fast with the copypaste lol

Comment (louis-bompart) [inline packages/cli/core/README.md:39]:
```suggestion
  - macOS: `~/Library/Caches/@coveo/cli`
```

File: packages/cli/core/README.md [hunk 3]
```diff
@@ -31,6 +31,28 @@ USAGE
...
+  - Windows: `%LOCALAPPDATA%\mycli`
+  - Can be overridden with `XDG_CACHE_HOME`
+- **configDir** - CLI config directory
+  - Unix: `~/.config/mycli`
+  - Windows: `%LOCALAPPDATA%\mycli`
+  - Can be overridden with `XDG_CONFIG_HOME`
+- **dataDir** - CLI data directory
+  - Unix: `~/.data/mycli`
+  - Windows: `%LOCALAPPDATA%\mycli`
+  - Can be overridden with `XDG_DATA_HOME`
```
Comment (louis-bompart) [inline packages/cli/core/README.md:50]:
```suggestion
  - Unix: `~/.cache/@coveo/cli`
  - Windows: `%LOCALAPPDATA%\@coveo\cli`
  - Can be overridden with `XDG_CACHE_HOME`
- **configDir** - CLI config directory
  - Unix: `~/.config/@coveo/cli`
  - Windows: `%LOCALAPPDATA%\@coveo\cli`
  - Can be overridden with `XDG_CONFIG_HOME`
- **dataDir** - CLI data directory
  - Unix: `~/.data/@coveo/cli`
  - Windows: `%LOCALAPPDATA%\@coveo\cli`
  - Can be overridden with `XDG_DATA_HOME`
```

**Review by y-lakhdar — APPROVED** [review]

**Pull request merged**